### PR TITLE
Schema 4.6

### PR DESCRIPTION
**Pull request by kaysiz**
## Purpose
Add new values for various attributes

- Addition of [Award](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/resourceTypeGeneral/#award) and [Project](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/resourceTypeGeneral/#project) to the [resourceTypeGeneral](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/resourceTypeGeneral/) controlled list values.
These values may be used in [10.a resourceTypeGeneral](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/resourcetype/#a) and other places where resourceTypeGeneral is used ([12.f resourceTypeGeneral](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/relatedidentifier/#f), [20.a relatedItemType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/relateditem/#a)).
- Addition of [CSTR](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relatedIdentifierType/#cstr) and [RRID](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relatedIdentifierType/#rrid) to the [relatedIdentifierType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relatedIdentifierType/) controlled list values.
These values may be used in [12.a relatedIdentifierType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/relatedidentifier/#a) and other places where relatedIdentifierType is used ([20.1.a relatedItemIdentifierType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/relateditem/#id3)).
- Addition of new [contributorType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/contributorType/): [Translator](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/contributorType/#translator)
This value may be used in [7.a contributorType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/contributor/#a) and other places where contributorType is used ([20.12.a contributorType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/properties/relateditem/#id19)).
- Addition of new [relationType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relationType/) pair: [HasTranslation](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relationType/#hastranslation) and [IsTranslationOf](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/relationType/#istranslationof)
- Addition of new [dateType](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/dateType/): [Coverage](https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/dateType/#coverage)

### Spec documents:
- [Metadata 4.6 - Technical Spec](https://docs.google.com/document/d/1bMPlgWoA2lsAcwzR81dSRvIxS9zOGYfwFTGyQtQbgEU/edit#heading=h.xic4e89kbl8a)
- [Metadata Schema Release 4.6](https://docs.google.com/document/d/1RllY4Q-_Ls9gHSfy9Iuwlctv0uV4E_S6Sg56rvkMt1o/)

closes: #887 #888 #886 #884 #883 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: cypress/fixtures/doi.xml
Comment (svogt0511) [inline cypress/fixtures/doi.xml:1]: Just a question.  Is there a reason why we use both schemaLocation for kernel-4 and kernel-4.6?  I thought that using kernel-4 would be the latest.

**Review by svogt0511 — COMMENTED** [review]
@kaysiz - A couple of comments in the code.  Also I left some comments at the vercel preview at https://bracco-rdckjb3ze-datacite.vercel.app/dois/10.82531%2Fta9w-p411.  Let me know if you don't see those comments.

File: app/components/doi-related-identifier.js [hunk 1]
```diff
@@ -43,13 +43,16 @@ const relationTypeList = [
   'Is original form of',
   'Is identical to',
   'Is collected by',
-  'Collects'
+  'Collects',
```
Comment (kaysiz) [inline app/components/doi-related-identifier.js:46]: @codycooperross @KellyStathis What order should we follow for this list?

File: cypress/fixtures/doi.xml
Comment (kaysiz) [reply cypress/fixtures/doi.xml:1]: @svogt0511 the first part is the namespace and the second is the url to the file that will validate the data

File: app/components/doi-contributor.js
[the remark this answers is not in the document]
Comment (kaysiz) [reply app/components/doi-contributor.js:1]: @svogt0511 https://datacite-metadata-schema.readthedocs.io/en/4.6_draft/appendices/appendix-1/contributorType/#translator the description gives an idea of what the contributorType group is

[the remark this answers is not in the document]
Comment (kaysiz) [reply app/components/doi-contributor.js:1]: In this case however `Transalator` can be both personal or organization. From the pattern it should probably just stay in the `contributorTypes` list. @codycooperross @KellyStathis can weigh in on this.

[the remark this answers is not in the document]
Comment (codycooperross) [reply app/components/doi-contributor.js:1]: It makes sense to me that a `Translator` could be either `Organizational` or `Personal`.

File: app/components/doi-related-identifier.js (hunk 1 above)
Comment (codycooperross) [reply app/components/doi-related-identifier.js:46]: In the past, I think this list has been additive, i.e. we just keep adding new values at the bottom :) What are some other options? Alphabetical, according to the list in the schema docs: https://datacite-metadata-schema.readthedocs.io/en/4.5/appendices/appendix-1/relationType/ —any others?

File: app/components/doi-contributor.js
[the remark this answers is not in the document]
Comment (svogt0511) [reply app/components/doi-contributor.js:1]: @kaysiz I will leave this for you to mark as resolved.

**Review by svogt0511 — APPROVED** [review]
Looks good to me.

**Comment by codycooperross** [comment]
@kaysiz Caught one more small issue—I'm getting this error when selecting a contributor type in RelatedItem:

```
Uncaught RangeError: Maximum call stack size exceeded
    at WeakMap.get (<anonymous>)
    at c.get (index.js:145:35)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
    at c.selectContributorType (doi-related-item-contributor.js:135:10)
```

Screen Shot 2024-11-26 at 3 09 52 PM

Otherwise, the CSTR issue looks fixed, and everything else looks great! If you could take a look at this, I think we're all set.

**Comment by codycooperross** [comment]
Looks like it was an existing bug, so I made a quick fix: 996597e If you could test that fix quickly, that would be great. Otherwise, all set :)

**Review by codycooperross — APPROVED** [review]

**Review by svogt0511 — APPROVED** [review]
Looks good.

**Pull request merged**